File: allocation-service/src/main/kotlin.com.taager.allocation/capacity/queries/application/usecases/GetProvinces.kt
package com.taager.allocation.capacity.queries.application.usecases
import com.taager.allocation.capacity.queries.application.contracts.ProvinceRepo
import com.taager.allocation.capacity.queries.application.models.ProvinceResponse
import com.taager.allocation.sharedkernel.application.UseCaseWithoutRequest
import org.springframework.stereotype.Service
@Service
class GetProvinces(val provinceRepo: ProvinceRepo): UseCaseWithoutRequest<List<ProvinceResponse>>() {
    override fun execute(): List<ProvinceResponse> {
        logger.debug("Get all provinces")
        return provinceRepo.getAllProvinces();
    }
}

File: Merchant-service/src/order-management/queries/application/usecases/province/GetProvinces.ts
import { ProvinceRepo } from '../../contracts/ProvinceRepo';
import { GetProvincesResponse } from '../../models/province/GetProvincesResponse';

export default class GetProvinces {
  private provincesRepo: ProvinceRepo;

  constructor(provincesRepo: ProvinceRepo) {
    this.provincesRepo = provincesRepo;
  }

  async execute(country: string): Promise<GetProvincesResponse[]> {
    return this.provincesRepo.getProvinces(country);
  }
}



